Release notes for 7.16.0 (#13436)
elastic/logstash: docs/static/releasenotes.asciidoc

* The use of the `JAVA_HOME` environment variable is targeted for removal in 8.0, please use `LS_JAVA_HOME` instead.

* Reminder that the Ruby Execution Engine is deprecated and will not be available in Logstash 8.
  The Java Execution Engine has been the default since 7.0 and reached substantial maturity in 7.10.

[[featured-7-16-0]]
==== New features and enhancements

* Added options for securing the Logstash HTTP API using TLS and/or HTTP Basic auth https://github.com/elastic/logstash/pull/13308[#13308].
  We're planning on following up on the feature set shipped in 7.16 for details please see https://github.com/elastic/logstash/issues/13196[#13196].

* Opting in to ECS compatibility per-pipeline (`pipeline.ecs_compatibility`) or globally has graduated from BETA.

* Introduce `LS_JAVA_HOME` environment variable https://github.com/elastic/logstash/pull/13204[#13204].
  Logstash launch scripts have been using `JAVA_HOME` to identify a JDK configured by the user,
  as that environment variable could be too generic Logstash now has a dedicated `LS_JAVA_HOME`.
  If `LS_JAVA_HOME` is not specified Logstash still looks for `JAVA_HOME`.

> * Logstash started supporting JDK 17 https://github.com/elastic/logstash/pull/13330[#13330]
>   and tests against Java 17 https://github.com/elastic/logstash/pull/13330[#13330]

Reviewer: ```suggestion
* Logstash added tests against Java 17 and started supporting JDK 17 https://github.com/elastic/logstash/pull/13330[#13330]
```

The complete list of supported operating systems and JVMs is available in the
https://www.elastic.co/support/matrix[support matrix].

[[ecs-7-16-0]]
===== Progress toward Elastic Common Schema (ECS)

In this release, we continued our efforts towards Elastic Common Schema (ECS):

* <<plugins-inputs-elasticsearch,elasticsearch input>> has sensible defaults with `ecs_compatibility`
* <<plugins-inputs-http_poller,http_poller input>> leverages ECS fields under `ecs_compatibility`
* <<plugins-codecs-avro,avro codec>> added `target` support
* <<plugins-codecs-es_bulk,ES bulk codec>> added `target` support

[[plugins-7-16-0]]
==== Plugins

*Avro Codec - 3.3.0*

* Add ECS support. Add target option and event.original https://github.com/logstash-plugins/logstash-codec-avro/pull/36[#36]

*Es_bulk Codec - 3.1.0*

* Add ECS support. Add `target` option https://github.com/logstash-plugins/logstash-codec-es_bulk/pull/20[#20]

*Aggregate Filter - 2.10.0*

> * new feature: add ability to generate new event during code execution (#116)

Reviewer: ```suggestion
* Feat: add ability to generate new event during code execution https://github.com/logstash-plugins/logstash-filter-aggregate/pull/116[#116]
```

*Elasticsearch Filter - 3.11.0*

* Feat: update Elasticsearch client to 7.14.0 https://github.com/logstash-plugins/logstash-filter-elasticsearch/pull/150[#150]

> 

Reviewer: ```suggestion
```

* Feat: add user-agent header passed to the Elasticsearch HTTP connection https://github.com/logstash-plugins/logstash-filter-elasticsearch/pull/152[#152]

*Azure_event_hubs Input - 1.4.0*

* Updated the minor version of Azure SDK and other dependencies to ensure users of this plugin get upstream fixes and improvements https://github.com/logstash-plugins/logstash-input-azure_event_hubs/pull/67[#67]

*Elasticsearch Input - 4.12.1*

> * Fixed too_long_frame_exception by passing scroll_id in the body https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/159[#159]
> 
> * Feat: Update Elasticsearch client to 7.14.0 https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/157[#157]
> 
> * Feat: add user-agent header passed to the Elasticsearch HTTP connection https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/158[#158]
> 
> * Feat: added ecs_compatibility + event_factory support https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/149[#149]

Reviewer: ```suggestion
* Fixed too_long_frame_exception by passing scroll_id in the body https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/159[#159]
* Feat: Update Elasticsearch client to 7.14.0 https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/157[#157]
* Feat: add user-agent header passed to the Elasticsearch HTTP connection https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/158[#158]
* Feat: added ecs_compatibility + event_factory support https://github.com/logstash-plugins/logstash-input-elasticsearch/pull/149[#149]
```

*Http_poller Input - 5.1.0*

* Add ECS support https://github.com/logstash-plugins/logstash-input-http_poller/pull/129[#129]

*Elasticsearch Output - 11.2.1*

> * Fix referencing Gem classes from global lexical scope https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1044[#1044]
> 
> * Added preflight checks on Elasticsearch https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1026[#1026]
> 
> * Feat: add `user-agent` header passed to the Elasticsearch HTTP connection https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1038[#1038]
> 

Reviewer: ```suggestion
* Fix referencing Gem classes from global lexical scope https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1044[#1044]
* Added preflight checks on Elasticsearch https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1026[#1026]
* Feat: add `user-agent` header passed to the Elasticsearch HTTP connection https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1038[#1038]

```

==== Performance improvements and notable issues fixed

* Add `pipeline.ordered` setting for docker image https://github.com/elastic/logstash/pull/13300[#13300]

**Updates to dependencies**

* Update JRuby to 9.2.20.1 https://github.com/elastic/logstash/pull/13468[#13468]

[[deprecated-7-16-0]]
==== Deprecations

* Logstash 8.0 will remove support for Java versions before 11, thus using Java 8 has been deprecated.

* The use of the `JAVA_HOME` environment variable is targeted for removal in 8.0, please use `LS_JAVA_HOME` instead.

* Reminder that the Ruby Execution Engine is deprecated and will not be available in Logstash 8.
  The Java Execution Engine has been the default since 7.0 and reached substantial maturity in 7.10.

[[featured-7-16-0]]
==== New features and enhancements

* Added options for securing the Logstash HTTP API using TLS and/or HTTP Basic auth https://github.com/elastic/logstash/pull/13308[#13308].
  We're planning on following up on the feature set shipped in 7.16 for details please see https://github.com/elastic/logstash/issues/13196[#13196].

* Opting in to ECS compatibility per-pipeline (`pipeline.ecs_compatibility`) or globally has graduated from BETA.

> * Introduce `LS_JAVA_HOME` environment variable https://github.com/elastic/logstash/pull/13204[#13204].
>   Logstash launch scripts have been using `JAVA_HOME` to identify a JDK configured by the user,
>   as that environment variable could be too generic Logstash now has a dedicated `LS_JAVA_HOME`.
>   If `LS_JAVA_HOME` is not specified Logstash still looks for `JAVA_HOME`.

Reviewer 2: ```suggestion
* Introduce `LS_JAVA_HOME` environment variable to replace `JAVA_HOME` https://github.com/elastic/logstash/pull/13204[#13204].
 Until 8.0, Logstash looks for `JAVA_HOME` If `LS_JAVA_HOME` is not specified.  (See <<deprecated-7-16-0>>.)
  
  We recommend using the bundled JDK unless you have a compelling reason to use a different one. 
```

Reviewer 2: This was confusing in light of the JAVA_HOME deprecation we listed earlier, so I tried to get the messaging in sync. Feel free to tweak.

* Logstash added tests against Java 17 and started supporting JDK 17 https://github.com/elastic/logstash/pull/13330[#13330]

The complete list of supported operating systems and JVMs is available in the
https://www.elastic.co/support/matrix[support matrix].

[[ecs-7-16-0]]
===== Progress toward Elastic Common Schema (ECS)

In this release, we continued our efforts towards Elastic Common Schema (ECS):

* <<plugins-inputs-elasticsearch,elasticsearch input>> has sensible defaults with `ecs_compatibility`
* <<plugins-inputs-http_poller,http_poller input>> leverages ECS fields under `ecs_compatibility`
* <<plugins-codecs-avro,avro codec>> added `target` support
* <<plugins-codecs-es_bulk,ES bulk codec>> added `target` support

> 

Reviewer 2: ```suggestion

Check out our https://github.com/elastic/logstash/issues/11635[progress toward ECS compatibility] in github issue https://github.com/elastic/logstash/issues/11635[#11635].

```

Reviewer 2: Hey, let's add this sentence, too, so that users can continue to track our overall progress toward ECS. 

Github suggestions are tricky for docs details such as spacing, etc.  Please make sure that the spacing works correctly if you choose to accept this suggection. :-)
